File: Make Human Unity/Assets/MakeHuman/Icons/KsSoft/Plugins/Protocol/t_EffectAttributeHeader.cs
//==============================================================================================
/*!
	@file  KsSoft Protocol
	
	(counter SJIS string 京)
*/
//==============================================================================================
using UnityEngine;
using System.Collections;
using System.Collections.Generic;

using KS;
//==========================================================================
/*!
	@brief	t_EffectAttributeHeader
*/
public class t_EffectAttributeHeader : ISerializable {
	//==========================================================================
	/*!
		@brief	Member
	*/
	public  const uint	MAGIC_NO = 4276944900;	// 4276944900
	public uint	m_magicNo;
	public t_EffectAttribute[]	m_aEffectAttribute;
	//==========================================================================
	/*!
		@brief	Constructor
	*/
	public t_EffectAttributeHeader() {
		clear();
	}
	//==========================================================================
	/*!
		@brief	Accessor
	*/
	public void clear() {
		m_magicNo = 4276944900;	// MAGIC_NO;
		m_aEffectAttribute = new t_EffectAttribute[0];
	}
	public bool read(CReadVariable cVariable) {
		try {
			m_magicNo = cVariable.getU32();
			{
				int n = (int) cVariable.getU16();
				if (n > 65535) {
					cVariable.error("array size error in m_aEffectAttribute");
					return false;
				}
				m_aEffectAttribute = new t_EffectAttribute [n];
				for(int i = 0;i < n;++i) {
					m_aEffectAttribute[i] = new t_EffectAttribute();
					if (!m_aEffectAttribute[i].read(cVariable)) return false;
				}
			}
		} catch (System.Exception e) {
			Debug.LogError(e);return false;
		}
		return true;
	}
	public bool write(CWriteVariable cVariable) {
		cVariable.put(m_magicNo);
		{	
				if (m_aEffectAttribute.Length > 65535) {
					cVariable.error("array size error in m_aEffectAttribute");
					return false;
				}
			int n = m_aEffectAttribute.Length;
			cVariable.put((ushort) n);
			for(int i = 0;i < n;++i) {
				if (!m_aEffectAttribute[i].write(cVariable)) return false;

			}
		}
		return true;
	}
};

File: Assets/MakeHuman/Icons/KsSoft/Editor/Multilingual/tools/protocol/t_EffectAttributeHeader.py
from CVariable import CReadVariable,CWriteVariable
from Vector import *
from t_EffectAttribute import	t_EffectAttribute
#/==========================================================================
#*!
#	@brief	t_EffectAttributeHeader
#/
class t_EffectAttributeHeader:
	#/==========================================================================
	#*!
	#	@brief	Member
	#/
	MAGIC_NO = 4276944900
	m_magicNo = 4276944900	#U32
	m_aEffectAttribute = []	#t_EffectAttribute
	#/==========================================================================
	#*!
	#	@brief	Constructor
	#/
	def __init__(self):
		self.clear()

	#/==========================================================================
	#*!
	#	@brief	Accessor
	#/
	def clear(self):
		self.m_magicNo = 4276944900
		self.m_aEffectAttribute = []

	def read(self,cVariable):
		self.m_magicNo = cVariable.getU32()
		n = cVariable.getU16()
		if (n > 65535) :
			cVariable.error('array size error in m_aEffectAttribute')
			return False
		self.m_aEffectAttribute = [None]*n
		for i in range(n):
			self.m_aEffectAttribute[i] = t_EffectAttribute()
			if (not self.m_aEffectAttribute[i].read(cVariable)):
				return False
		return True

	def write(self,cVariable):
		cVariable.putU32(self.m_magicNo)
		if (len(self.m_aEffectAttribute) > 65535) :
			cVariable.error('array size error in m_aEffectAttribute')
			return False
		n = len(self.m_aEffectAttribute)
		cVariable.putU16(n)
		for i in range(n):
			if (not self.m_aEffectAttribute[i].write(cVariable)):
				return False
		return True

